targets/targets.json

```diff
@@ -630,6 +630,19 @@
         "release_versions": ["2", "5"],
         "device_name": "MK82FN256xxx15"
     },
+    "LPC54114": {
```

Comment:
This naming scheme is following what was done for other Freescale platforms. Also this name matches what is used in the mbed database.